src/crimson/os/seastore/device.h

```diff
@@ -85,6 +85,9 @@ class Device {
   using mkfs_ertr = access_ertr;
   using mkfs_ret = mkfs_ertr::future<>;
   virtual mkfs_ret mkfs(device_config_t) = 0;
+  virtual mkfs_ret shard_mkfs(device_config_t) {
+    return mkfs_ertr::now();
+  }
```

Comment:
I think the mkfs paths in SeaStore is still too complicated.

Like `FuturizedStore` vs `FuturizedShardStore`, I think it is reasonable to split into `Device` and `ShardedDevice`, so that `SeaStore` only needs to worry about global device initializations like `Device::mkfs()`, `mount()`, `start()` and `stop()`, and after that, `SeaShardStore` can get its own `ShardedDevice` through `Device::get_sharded_device()`.

There are other reasons to split the `Device` class, for example, the global member like `superblock` should be operated by the `BlockSegmentManager` rather than `ShardedBlockSegmentManager`. And I suppose there will be more implementations in `BlockSegmentManager` to support cross-shard dynamic partition in the future.

Reply:
It's really not clear what the intended distinction between mkfs and shard_mkfs is here.

Reply:
Seems to me that mkfs() and shard_mkfs() are similar to CyanStore::mkfs() and CyanShardStore::mkfs(), that mkfs() is to mkfs at device-level (e.g. superblock), and shard_mkfs() is to mkfs at shard-level (e.g. shard-tracker).

This is why I suggest to hide the entire mkfs process inside the device, so that seastore mkfs for the device will look much simpler.